node/src/components/storage.rs

```diff
@@ -804,7 +923,55 @@ impl Storage {
         }))
     }
 
-    // Retrieves a block header to handle a network request.
+    /// Retrieves a block by hash.
+    pub fn read_block(&self, block_hash: &BlockHash) -> Result<Option<Block>, Error> {
+        self.get_single_block(&mut self.env.begin_ro_txn()?, block_hash)
+    }
+
+    /// Writes a block to storage, updating indices as necessary
+    pub fn write_block(&mut self, block: &Block) -> Result<bool, Error> {
+        // Validate the block prior to inserting it into the database
+        block.verify()?;
```

Comment:
This might already be taken care of on `feat-fast-sync`, I think the PR for making `Block::verify` private was changing that.

Reply:
Yeah: https://github.com/casper-network/casper-node/pull/1928/files#diff-c18aa5de121b5c6ebe423bfa9f6d00d885902cdcb9f640abd8eec1b1da4a0ecdL989

It's just something I haven't ported, partially because it hadn't been merged yet then. If you think I should port this as well in this PR, I can try to do so, but I think it can be left for the future as well.

Reply:
I think we should have the `Block::verify`-private refactor in a separate PR (WDYT @afck)?